applications/ukf_mixed_precision.cpp

```
@@ -425,6 +468,62 @@ double significand_bits_at_one() {
+
+template <typename S>
+Pivots<S> trailing_pivots(const double Pd[N][N]) {
+    const double ad = Pd[0][0], bd = Pd[0][1], cd = Pd[1][1];
+    Pivots<S> p;
+    p.exact = cd - bd * bd / ad;
+
+    // Same gate the main table uses. Without it an out-of-range entry becomes
+    // inf and every column below prints inf - inf = nan.
+    if (!representable<S>(Pd)) {
+        p.in_range = false;
+        return p;
```

Cppcheck (2.21.0)

[error] Uninitialized variables

(uninitvar)